### PR TITLE
add session report wiki tag

### PR DESCRIPTION
**Pull request by elhardoum**
I waited on sending a PR because this wasn't fully completed yet. For the time being, it uses sample fake data, and not the report you've sent yesterday.

Will be working on implementing the mocks provided to me into the report next.

### CODE REVIEW

File: src/services/api.js [hunk 1]
```diff
@@ -114,28 +114,6 @@ async function getDownload(props, file) {
     })
 }
```
Comment (RotateAt60MPH) [inline src/services/api.js:116]: There was an existing call in here.  What was the reason for moving it?

File: src/utils/PlayerState.js [hunk 2]
```diff
@@ -169,6 +170,20 @@ class PlayerState {
   static GetNodesVisited(keyPrefix, defaultValue = null) {
     return persistantStorage.get(keyPrefix, KeyConstants.VISIT_ONCE_NODE_LIST, defaultValue);
   }
+
+  static SetSessionReport(keyPrefix, obj) {
```
Comment (RotateAt60MPH) [inline src/utils/PlayerState.js]: Why is the session report persisted to storage?  It really only has a life during that one view...

File: src/components/WikiTags/WikiTags.js [hunk 3]
```diff
@@ -176,6 +178,36 @@ const combineStyles = (...styles) => {
   };
 };
 
+async function getSessionReport(props, contextId) {
```
Comment (RotateAt60MPH) [inline src/components/WikiTags/WikiTags.js]: For consistency purposes, this could be in api.js.

File: src/components/WikiTags/Report/Report.js [hunk 4]
```diff
@@ -32,11 +79,7 @@ class OlabReportTag extends React.Component {
         );
       }
 
-      return (
-        <>
-          <b>[[REPORT]]</b>
-        </>
-      );
+      return (<Report {...this.props} report={report} />);
```
Comment (RotateAt60MPH) [inline src/components/WikiTags/Report/Report.js]: Could you please tweak this to use only class components?  And one component per file.

File: src/components/WikiTags/WikiTags.js [hunk 5]
```diff
@@ -176,6 +178,36 @@ const combineStyles = (...styles) => {
...
+
+  if ( !0 ) {
```
Comment (RotateAt60MPH) [inline src/components/WikiTags/WikiTags.js]: This can be removed?

File: src/components/WikiTags/Report/Report.js [hunk 6]
```diff
@@ -2,28 +2,75 @@
...
+
+    if (report && !disableCache) {
+      log.debug('using cached session report data', report);
+      return report;
+    }
+
+    // fetch report data from the api endpoint
+    report = await getSessionReport(this.props.props, contextId);
+
+    if (report && !disableCache) {
```
Comment (RotateAt60MPH) [inline src/components/WikiTags/Report/Report.js]: Shouldn't need to cache the results, I'm thinking.  See other comment.

File: src/services/api.js (hunk 1 above)
Comment (elhardoum) [reply src/services/api.js:116]: Not sure what's being quoted here.

File: src/components/WikiTags/WikiTags.js (hunk 3 above)
Comment (elhardoum) [reply src/components/WikiTags/WikiTags.js]: You asked me to move that to the WikiTags.js component, in our meeting last month. I'll move it back, no problem.

File: src/utils/PlayerState.js (hunk 2 above)
Comment (elhardoum) [reply src/utils/PlayerState.js]: Just thought it would be consistent, the way most of the other data is cached. I'll get rid of that.

**Pull request merged**